library/std/src/sys/windows/compat.rs

```diff
@@ -21,9 +21,48 @@
 
 use crate::ffi::{c_void, CStr};
 use crate::ptr::NonNull;
-use crate::sync::atomic::{AtomicBool, Ordering};
+use crate::sync::atomic::Ordering;
 use crate::sys::c;
 
+// This uses a static initializer to preload some imported functions.
+// The CRT (C runtime) executes static initializers before `main`
+// is called (for binaries) and before `DllMain` is called (for DLLs).
+//
+// It works by contributing a global symbol to the `.CRT$XCT` section.
+// The linker builds a table of all static initializer functions.
+// The CRT startup code then iterates that table, calling each
+// initializer function.
+//
+// NOTE: User code should use ".CRT$XCU" to guarantee they run after the
+// standard library's initializers.
```

Comment:
Are we comfortable with documenting this as a guarantee? It feels a little odd to have this snippet in a random comment inside std, though I doubt we'd want to surface this in std docs either.

I would feel a little more comfortable if we rephrased to something like "Note: User code may rely on this and use .CRT$XCU to run *after* std's initializer. If you're reading this and would like a guarantee here, please file an issue for discussion; currently we don't guarantee any functionality before main."

Reply:
Ah, that sounds better. You're right, I'm not really looking to provide any new guarantees. Just trying to dissuade people from copying exactly what std does, which would defeat the purpose.